auth0/v3/authentication/token_verifier.py

```diff
@@ -340,3 +339,130 @@ def _verify_payload(self, payload, nonce=None, max_age=None, organization=None):
+            # noinspection PyUnresolvedReferences
+            ustr = unicode
+        except NameError:
+            ustr = str
+
+        # Issuer
+        if 'iss' not in payload or not isinstance(payload['iss'], (str, ustr)):
+            raise TokenValidationError('Issuer (iss) claim must be a string present in the access token')
+        if payload['iss'] != self.iss:
+            raise TokenValidationError(
+                'Issuer (iss) claim mismatch in the access token; expected "{}", '
+                'found "{}"'.format(self.iss, payload['iss']))
+
+        # Subject
+        if 'sub' not in payload or not isinstance(payload['sub'], (str, ustr)):
+            raise TokenValidationError('Subject (sub) claim must be a string present in the access token')
+
+        # Audience
+        if 'aud' not in payload or not (isinstance(payload['aud'], (str, ustr)) or isinstance(payload['aud'], list)):
+            raise TokenValidationError(
+                'Audience (aud) claim must be a string or array of strings present in the access token')
+
+        if isinstance(payload['aud'], list) and self.aud not in payload['aud']:
+            payload_audiences = ", ".join(payload['aud'])
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the access token; expected "{}" but was '
+                'not one of "{}"'.format(self.aud, payload_audiences))
+        elif isinstance(payload['aud'], (str, ustr)) and payload['aud'] != self.aud:
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the access token; expected "{}" '
+                'but found "{}"'.format(self.aud, payload['aud']))
+
+        # --Time validation (epoch)--
+        now = self._clock or time.time()
+        leeway = self.leeway
+
+        # Expires at
+        if 'exp' not in payload or not isinstance(payload['exp'], int):
+            raise TokenValidationError('Expiration Time (exp) claim must be a number present in the access token')
+
+        exp_time = payload['exp'] + leeway
+        if now > exp_time:
+            raise TokenValidationError(
+                'Expiration Time (exp) claim error in the access token; current time ({}) is '
+                'after expiration time ({})'.format(now, exp_time))
+
+        # Issued at
+        if 'iat' not in payload or not isinstance(payload['iat'], int):
+            raise TokenValidationError('Issued At (iat) claim must be a number present in the access token')
+
+        # Scopes
+        if scopes:
+            if 'scopes' not in payload or not isinstance(payload['scopes'], str):
+                raise TokenValidationError('No scopes claim found in token.')
+
+            for scope in scopes.split(' '):
+                if scope not in payload['scopes'].split(' '):
+                    msg = "You don't have access to this resource"
+                    raise Auth0Error(status_code=401, error_code=401, message=msg)
```

Comment:
I'm not sure this is the correct error status code or error code and message for this case

Reply:
The OAuth spec specifies a `400` reqiest for an invalid scope error, something like: 

```
HTTP/1.1 400 Bad Request
     Content-Type: application/json;charset=UTF-8
     Cache-Control: no-store
     Pragma: no-cache

     {
       "error":"invalid_scope"
     }
```

https://datatracker.ietf.org/doc/html/rfc6749#section-5.2

Reply:
While digging deeper in the OAuth 2.0 spec, it seems the above `400` needs to be thrown on requesting an access token with in invalid scopes, not when validating an access token's scopes.

The spec does not specify how to handle this scenario

> If a resource access request fails, the resource server SHOULD inform
   the client of the error.  While the specifics of such error responses
   are beyond the scope of this specification, this document establishes
   a common registry in Section 11.4 for error values to be shared among
   OAuth token authentication schemes. 

The `express-oauth2-bearer` lib [throws](https://github.com/auth0/express-oauth2-bearer/blob/master/lib/errors.js#L22) a `403`, let's use that to stay consistent.